### ISSUE TITLE
Not rolling back the transaction properly on GitHub Apps Update

### ISSUE THREAD

**Issue by coverbeck** [dockstore/dockstore#4434]
**Describe the bug**
If there is an error adding a new workflow version via GitHub apps (didn't test updating an existing version), the version can still get created. It depends where the error is.

**To Reproduce**
1. Add a new version to an existing .dockstore.yml workflow with the steps in #4431
2. Push the branch
3. See an error in the logs/locally
4. Refresh My Workflows -- the new version is there!
5. Note how the info tab does not have the README displayed (if there is one). See below for why.

**Expected behavior**
The version should not get created. Although in this case, given #4431, it's probably just as well that it is for now.

**Additional context**
I'm pretty sure this is due to the block starting [here](https://github.com/dockstore/dockstore/blob/2b4803cbfe069c908baaa59ec3460ae2a5141c89/dockstore-webservice/src/main/java/io/dockstore/webservice/resources/AbstractWorkflowResource.java#L348). The `clear()` only clears items that [have not been persisted](https://stackoverflow.com/questions/8118436/org-hibernate-session-clear-considered-harmful) (hard to find in official doc). As we create a version, we persist the workflow version, the sourcefiles, etc. With #4431 it fails trying to set the version metadata. So everything up to version metadata gets saved [here](https://github.com/dockstore/dockstore/blob/2b4803cbfe069c908baaa59ec3460ae2a5141c89/dockstore-webservice/src/main/java/io/dockstore/webservice/resources/AbstractWorkflowResource.java#L353). That's why the README is not displayed.

TLDR; I think we need to explicitly call `rollback` instead of or as well as `clear`, and start a new transaction.

[Webservice](https://github.com/dockstore/dockstore/releases/tag/1.11.5) - 1.11.5

[UI](https://github.com/dockstore/dockstore-ui2/releases/tag/2.8) - 2.8

[Compose Setup](https://github.com/dockstore/compose_setup/releases/tag/1.11) - 1.11

[Deploy](https://github.com/dockstore/dockstore-deploy/releases/tag/1.11.0) - 1.11.0

┆Issue is synchronized with this [Jira Story](https://ucsc-cgl.atlassian.net/browse/DOCK-1896)
┆Issue Number: DOCK-1896
┆Sprint: Sprint 70- Whale shark
┆Issue Type: Story
